src/main/scala/com/fulcrumgenomics/bam/ZipperBams.scala

```diff
@@ -0,0 +1,242 @@
+    |is "Consensus" which contains all the per-base consensus tags produced by fgbio consensus callers.
+    |
+    |By default the mapped BAM is read from standard input (stdin) and the output BAM is written to standard
+    |output (stdout). This can be changed using the `--input/-i` and `--output/-o` options.
+    |
+    |By default the output BAM file is emitted in the same order as the input BAMs.  This can be overridden
+    |using the `--sort` option, though in practice it may be faster to do the following:
+    |
+    |```
+    |fgbio --compression 0 ZipperBams -i mapped.bam -u unmapped.bam -r ref.fa | samtools sort -@ $(nproc)
+    |```
+  """)
+class ZipperBams
+( @arg(flag='i', doc="Mapped SAM or BAM.") val input: PathToBam = Io.StdIn,
+  @arg(flag='u', doc="Unmapped SAM or BAM.") val unmapped: PathToBam,
+  @arg(flag='r', doc="Path to the reference used in alignment.") val ref: PathToFasta,
+  @arg(flag='o', doc="Output SAM or BAM file.") val output: PathToBam = Io.StdOut,
+  @arg(doc="Tags to remove from the mapped BAM records.", minElements=0)
+  val tagsToRemove: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(doc="Set of optional tags to reverse on reads mapped to the negative strand.", minElements=0)
+  val tagsToReverse: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(doc="Set of optional tags to reverse complement on reads mapped to the negative strand.", minElements=0)
+  val tagsToRevcomp: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(flag='s', doc="Sort the output BAM into the given order.") val sort: Option[SamOrder] = None,
+  @arg(flag='b', doc="Buffer this many read-pairs while reading the input BAMs.") val buffer: Int = 5000
+) extends FgBioTool {
+  import ZipperBams._
+
+  override def execute(): Unit = {
+    val unmappedSource = SamSource(unmapped)
+    val mappedSource   = SamSource(input)
+    checkSort(unmappedSource, unmapped, "unmapped")
+    checkSort(mappedSource,  input, "input")
+
+    val dict    = SAMSequenceDictionaryExtractor.extractDictionary(this.ref)
+    val header  = buildOutputHeader(unmappedSource.header, mappedSource.header, this.sort, dict)
+    val out     = SamWriter(output, header, sort=this.sort)
+    val tagInfo = TagInfo(remove=tagsToRemove, reverse=tagsToReverse, revcomp=tagsToRevcomp)
+
+    if (tagInfo.remove.nonEmpty)  logger.info(s"Tags for removal from mapped BAM: ${tagInfo.remove.mkString(", ")}")
+    if (tagInfo.reverse.nonEmpty) logger.info(s"Tags being reversed: ${tagInfo.reverse.mkString(", ")}")
+    if (tagInfo.revcomp.nonEmpty) logger.info(s"Tags being reverse complemented: ${tagInfo.revcomp.mkString(", ")}")
+
+    val unmappedIter = templateIterator(unmappedSource, bufferSize=buffer)
+    val mappedIter   = templateIterator(mappedSource, bufferSize=buffer)
+
+    unmappedIter.foreach { template =>
+      if (mappedIter.hasNext && mappedIter.head.name == template.name) {
```

Comment:
Happy to keep your version for speed, but this also works:
```suggestion
      if (mappedIter.headOption.exists(_.name == template.name)) {
```

Reply:
I'll keep mine, thanks!